File: backend/app/schemas/objection.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime


class ObjectionCreate(BaseModel):
    answer_log_id: str = Field(max_length=36)
    objection_reason: str = Field(min_length=1, max_length=5000)


class ObjectionResponse(BaseModel):
    objection_id: str
    status: str


class ObjectionDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    quiz_session_id: str
    quiz_item_id: str
    answer_log_id: str
    objection_reason: str
    status: str
    review_result: dict | None = None
    decided_at: datetime | None = None
    decided_by: str | None = None
    created_at: datetime

File: frontend/src/types/objection.ts
export interface ObjectionCreate {
  answer_log_id: string;
  objection_reason: string;
}

export interface ObjectionResponse {
  objection_id: string;
  status: string;
}

export interface ObjectionDetail {
  id: string;
  quiz_session_id: string;
  quiz_item_id: string;
  answer_log_id: string;
  objection_reason: string;
  status: string;
  review_result: Record<string, unknown> | null;
  decided_at: string | null;
  decided_by: string | null;
  created_at: string;
}
